security/kernel-lockdown-tests.py

```diff
@@ -0,0 +1,96 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2023 IBM.
+# Author: Nageswara R Sastry <rnsastry@linux.ibm.com>
+
+import os
+from avocado import Test
+from avocado.utils import distro, genio, linux_modules, process
+
+
+class kernelLockdown(Test):
+    """
+    Kernel Lockdown tests for Linux
+    :avocado: tags=privileged,security,lockdown
+    """
+
+    def setUp(self):
+        '''
+        Kernel lockdown and Guest Secure boot support is available only on P10.
+        '''
+        self.distro_version = distro.detect()
+        val = genio.read_file("/proc/cpuinfo")
+        if 'POWER10' not in val:
+            self.cancel("Power10 LPAR is required to run this test.")
+        # Checking whether lockdown enabled or not.
+        self.lockdown_file = "/sys/kernel/security/lockdown"
+        lockdown = genio.read_file(self.lockdown_file).rstrip('\t\r\n\0')
+        index1 = int(lockdown.index('[')) + 1
+        index2 = int(lockdown.index(']'))
+        sys_lockdown = lockdown[index1:index2]
+        if sys_lockdown == 'none':
+            self.cancel("Lockdown not enabled, can't execute test(s)")
+        # Checking the Guest Secure Boot enabled or not.
+        cmd = "lsprop  /proc/device-tree/ibm,secure-boot"
+        output = process.system_output(cmd, ignore_status=True).decode()
+        if '00000002' not in output:
+            self.cancel("Secure boot is not enabled.")
+        # List required for kernel configuration check.
+        self.no_config = []
+
+    def _check_kernel_config(self, config_option):
+        # Helper function to check kernel config options
+        ret = linux_modules.check_kernel_config(config_option)
+        if ret == linux_modules.ModuleConfig.NOT_SET:
+            self.no_config.append(config_option)
+
+    def test_check_kernel_config(self):
+        # Checking the required kernel config options for lockdown
+        self._check_kernel_config('CONFIG_SECURITY_LOCKDOWN_LSM')
+        self._check_kernel_config('CONFIG_SECURITY_LOCKDOWN_LSM_EARLY')
+        self._check_kernel_config('CONFIG_LSM')
+        if self.no_config:
+            self.fail("Config options not set are: %s" % self.no_config)
+
+    def _runLockdownCommand(self, cmd, msg):
+        # Run the command and looking for the expected error from output
+        output = process.run(cmd, shell=True, ignore_status=True)
+        if 'Operation not permitted' not in output.stderr_text:
+            self.fail(msg)
+
+    def test_lockdown_none(self):
+        # Try changing the lockdown value to 'none'
+        cmd = "echo 'none' > %s" % self.lockdown_file
```

Comment:
can we have this as python way ? even we can use the utility as well like 

genio.write_one_line(self.lockdown_file, "none")

Reply:
ah we are running the command in another function, need to streamline this may be we can try/expect for all file operation as we should avoid running bash command as much possible in python